diff --git a/src/INS_filt3.py b/src/INS_filt3.py
index ac9a371..0aab8ee 100644
--- a/src/INS_filt3.py
+++ b/src/INS_filt3.py
@@ -19,7 +19,8 @@ for line in f:
 	if (int(depth_l[0]) + int(depth_l[1]))/2 < min_depth:
 #		print("depth", (int(depth_l[0]) + int(depth_l[1]))/2, line)
 		continue
-
+	if int(rep_l1[1]) == 0 or int(rep_l2[1]) == 0:
+		continue
 	if line_l[12] == "-" and (rep_l1[1] and int(rep_l1[0])/int(rep_l1[1]) < min_prop_lowq_reads) and (rep_l2[1] and int(rep_l2[0])/int(rep_l2[1]) < min_prop_lowq_reads) and int(line_l[4]) >= min_support_read:
 #		print(line_l[12], int(rep_l1[0])/int(rep_l1[1]), int(rep_l2[0])/int(rep_l2[1]), line_l[4])
 		print(line)
diff --git a/src/SB_CH2.py b/src/SB_CH2.py
index 395b3d7..df9b52e 100644
--- a/src/SB_CH2.py
+++ b/src/SB_CH2.py
@@ -59,7 +59,7 @@ total_num_read = len(read_name)
 for line in f:
 	if line[0] == "@":
 		read_name_fastq = (line.split(" "))[0]
-		read_name_fastq = read_name_fastq.replace("@", "")
+		read_name_fastq = read_name_fastq.replace("@", "").replace("\n", "")
 		if read_name_fastq in read_name:
 			line = line.replace("\n", "")
 			read_name[read_name_fastq] = line
@@ -121,6 +121,8 @@ for line in f:
 #@e22a5408-8e07-4b32-bf9a-e58c5c957654 runid=38469a38d9ea3789b624bd568739d7e40ba7c717 read=214 ch=383 start_time=2017-06-27T08:45:44Z
 			read_name1_all = read_name[breakpoint_info[i][0]]
 			read_name2_all = read_name[breakpoint_info[j][0]]
+			if read_name1_all == 1: continue
+			if read_name2_all == 1: continue
 			read_name1_all_l = read_name1_all.split(" ")
 			read_name2_all_l = read_name2_all.split(" ")
 #			read1 = int(read_name1_all_l[2].replace("read=", ""))
diff --git a/src/VAF.INS.py b/src/VAF.INS.py
index 4b36e01..4338e29 100644
--- a/src/VAF.INS.py
+++ b/src/VAF.INS.py
@@ -32,6 +32,8 @@ for line in f:
 
 	depth1 = 0
 	low_mq_read1 = 0
+	if int(line_l[1]) <= 100:
+		continue
 #	print(line_l[0], int(line_l[1]) - 100, int(line_l[1]) - 99)
 	for read in bamfile.fetch(line_l[0], int(line_l[1]) - 100, int(line_l[1]) - 99):
 #		print(read)
diff --git a/src/somatic_DEL_filt.py b/src/somatic_DEL_filt.py
index 16ea20c..6ac4a6b 100644
--- a/src/somatic_DEL_filt.py
+++ b/src/somatic_DEL_filt.py
@@ -247,6 +247,9 @@ for line in f:
 		if mq_bq_found == 0:
 			strange_read.append(j)
 
+	if len(breakpoint_info) == 0:
+		continue
+
 	breakpoint_start_l = []
 	breakpoint_end_l = []
 	for breakpoint_info_tmp in breakpoint_info:
diff --git a/src/somatic_SB_CH.py b/src/somatic_SB_CH.py
index a8d63cc..eda2966 100644
--- a/src/somatic_SB_CH.py
+++ b/src/somatic_SB_CH.py
@@ -81,7 +81,7 @@ total_num_read = len(read_name)
 for line in f:
 	if line[0] == "@":
 		read_name_fastq = (line.split(" "))[0]
-		read_name_fastq = read_name_fastq.replace("@", "")
+		read_name_fastq = read_name_fastq.replace("@", "").replace("\n", "")
 		if read_name_fastq in read_name:
 			line = line.replace("\n", "")
 			read_name[read_name_fastq] = line
@@ -143,8 +143,12 @@ for line in f:
 #@e22a5408-8e07-4b32-bf9a-e58c5c957654 runid=38469a38d9ea3789b624bd568739d7e40ba7c717 read=214 ch=383 start_time=2017-06-27T08:45:44Z
 			read_name1_all = read_name[breakpoint_info[i][0]]
 			read_name2_all = read_name[breakpoint_info[j][0]]
+			if read_name1_all == 1: continue
+			if read_name2_all == 1: continue
 			read_name1_all_l = read_name1_all.split(" ")
 			read_name2_all_l = read_name2_all.split(" ")
+			if len(read_name1_all_l) < 3: continue
+			if len(read_name1_all_l) < 3: continue
 			read1 = int(read_name1_all_l[2].replace("read=", ""))
 			read2 = int(read_name2_all_l[2].replace("read=", ""))
 
